javatests/arcs/android/storage/service/StorageServiceManagerTest.kt

```diff
@@ -100,43 +117,45 @@ class StorageServiceManagerTest {
         val manager = buildManager()
         val deferredResult = DeferredResult(coroutineContext)
 
-        val updatedData = Job()
-        handle.onUpdate { updatedData.complete() }
-
         manager.clearDataBetween(2,2, deferredResult)
 
         assertThat(deferredResult.await()).isTrue()
-        updatedData.join()
-        withContext(handle.dispatcher) {
-            assertThat(handle.fetchAll()).containsExactly(entity1, entity3)
+
+        // Create a new handle (with new Entity manager) to confirm data is gone from storage.
+        if(allRemoved) {
+            assertThat(createCollectionHandle(storageKey).fetchAll()).isEmpty()
+        } else {
+            assertThat(createCollectionHandle(storageKey).fetchAll())
+                .containsExactly(entity1, entity3)
         }
```

Comment:
Consider creating the handle, then wrapping the assertions within `withContext(handle.dispatcher)`.